### PR TITLE
DOCS Rework documentation to have clear user guide and dev docs.

### PR DESCRIPTION
**Pull request by camfindlay**
As per the supported module standard we should be making user guide related docs clear. This PR moves documentation relevant only to CMS users into the correct format. I have also added some new docs to make them more complete (I'm sure more could be added too).

Note I have included optional short user docs about comments and widgets here as the modules that provide them do not have user guides at all. I think it useful to maintain these here under this module rather than further split the user experience for reading the docs. We are also looking into better including  these supported modules in the userhelp website. This new format goes a way towards making that possible.

### CODE REVIEW

**Comment by camfindlay** [comment]
thanks for the peer. Will tidy up tomorrow.

**Comment by camfindlay** [comment]
@spiritlevel try this now I will leave the backtick code blocks for now this is acceptable GFMD format.

**Comment by camfindlay** [comment]
hold off merging, just working out a convention as to whether we use "user-guide" or "userguide" as the folder name.

**Comment by camfindlay** [comment]
@tractorcow this can go in like this. Can deal with any naming conventions later.

File: docs/en/index.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Blog documentation
+
+## Development
+ * [Configuring blog posts in Sitetree](configuring-blog-posts.md)
```
Comment (tractorcow) [inline docs/en/index.md]: Does docsviewer work with .md extensions now?: P

**Comment by tractorcow** [comment]
I actually like that you've dumped the "_images is in root" convention. I don't think it really worked in practice with deeply nested docs.

File: docs/en/configuring-blog-posts.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+## Configuring blog posts in SiteTree
+
+Because your blog is part of the SiteTree, usage is the same as any other page.
+
+By default, blog posts don't appear in the SiteTree, to avoid clutter. Instead they appear inside your blog as a GridField.
+
+![](docs/en/_images/blog-post-management.png)
```
Comment (tractorcow) [inline docs/en/configuring-blog-posts.md]: Links don't work with docs/en in them

File: docs/en/index.md (hunk 1 above)
Comment (camfindlay) [inline docs/en/index.md]: yes. this also allows the docs to function in the github repo.

File: docs/en/configuring-blog-posts.md (hunk 2 above)
Comment (camfindlay) [inline docs/en/configuring-blog-posts.md]: thanks, I'll fix this.

**Comment by camfindlay** [comment]
How's that @tractorcow ?

**Comment by tractorcow** [comment]
Good with me. Someone else can review and merge though, in case someone begs to differ. ;)

**Comment by tractorcow** [comment]
@assertchris or @micmania1 want to approve this?

**Comment by assertchris** [comment]
See https://groups.google.com/forum/#!topic/silverstripe-dev/GrSOZZPgEIk

**Comment by camfindlay** [comment]
It's ~ 50/50 split on the naming I think we should allow either (not to mention other modules are now using "userguide" directory now anyway.

**Comment by camfindlay** [comment]
@tractorcow this should be good to merge

**Pull request merged**